polaris-migrator/src/migrations/react-replace-text-components/steps/replace-other.ts

```diff
@@ -52,152 +52,164 @@ const components = {
     as: 'span',
     variant: 'bodySm',
   },
-} as const;
+};
 
-/**
- * Replace <Heading>, <Subheading>, <Caption>, and <VisuallyHidden> with the <Text> component
- */
-export function replaceOther<NodeType = ASTNode>(
+function replaceOtherComponent(
   j: JSCodeshift,
-  source: Collection<NodeType>,
-  options: MigrationOptions,
+  source: Collection,
+  componentName: string,
+  relative: boolean,
 ) {
-  Object.entries(components).forEach(
-    ([componentName, {validElements, defaultElement, variant, as}]) => {
-      const sourcePaths = normalizeImportSourcePaths(j, source, {
-        relative: options.relative,
-        from: componentName,
-        to: 'Text',
-      });
-
-      if (!sourcePaths) return;
+  const {variant, as, validElements, defaultElement} = (components as any)[
```

Comment:
You may be able to remove the `as any` assertion by updating the `componentName` type from `string` to `keyof typeof components`.